### ISSUE TITLE
encodeDefaults = false does not work when using inheritance for Serializable classes

### ISSUE THREAD

**Issue by truestep** [Kotlin/kotlinx.serialization#1511]
**To Reproduce**
while [this](https://github.com/Kotlin/kotlinx.serialization/issues/987#issuecomment-677745895) example works

this one doesn't
```

@Serializable
open class Project2(var name: String) {
    var renamedTo: String? = null
    var level: Int = 100
}

@Serializable
class Project3() : Project2(name = "abcd") {
    var user: String? = null
}

fun test2(){
    val json = Json{ encodeDefaults = false }
    val data = Project3()
    println(json.encodeToString(Project3.serializer(), data))
}

// I/System.out: {"name":"abcd","renamedTo":null,"level":100}
```

**Expected behavior**
In my case I'm interested to have the 
`I/System.out: {"name":"abcd"}`

// also this code worked like 3-4 moth ago, but now, after I updated to latest libs it has stopped

**Environment**
 - Kotlin version: 1.5.0
 - Library version: 1.5.0
 - Kotlin platforms: JVM
 - Gradle version: 6.7.1

**Comment by altavir** [comment]
Duplicate of #1467

**Comment by sandwwraith** [comment]
As a temporary workaround, you can use old JVM compiler instead of IR one. As it is stated in the original issue, this will be fixed in 1.5.20

**Comment by truestep** [comment]
When will the 1.5.20 going to be released for Android studio? 
How to switch to old JVM compiler in Android Studio?
